tests/integration_rpc_server.rs

```diff
@@ -0,0 +1,55 @@
+#[cfg(test)]
+use assert_cmd::Command;
+
+#[test]
+fn deploy_and_invoke_contract_against_rpc_server() {
+    // This test assumes a fresh standalone network rpc server on port 8000
```

Comment:
We need a way to run tests without running tests that require a running a stellar-core.

I see a couple ways we could do that:
- Have an env variable that provides the base URL of the RPC server. If it is set, use that. If it is not set, use `Command::cargo_bin` to run `soroban serve`, and then point to it. This has some huge advantages:
   - We can run tests twice, once with the serve, once with the real rpc server. Two tests for the price of one.
   - Random dev can run the tests without the heavy rpc server / stellar-core. We maintain agility.
   - People running the rpc server locally on a different port can do so.

- Add a cfg ignore to the tests `#[cfg_attr(not(feature = "tests_e2e_rpc_server"), ignore)]`. This has the advantages that the tests are always compiled, and when running tests a line will be outputted saying a test is being skipped. This is similar to "Skip" in Go tests.

- Add a cfg to the top of the file `#![cfg(feature = "tests_e2e_rpc_server")]`. This more convenient, but the tests won't compile at all unless the feature is enabled.

I prefer the first option because it involves us always running the tests in some form. Wdyt?

Reply:
> We need a way to run tests without running tests that require a running a stellar-core

For this particular test, I disagree. The intention is to have an end to end integration test that checks everything works.

Reply:
> * Have an env variable that provides the base URL of the RPC server. If it is set, use that. If it is not set, use `Command::cargo_bin` to run `soroban serve`, and then point to it.

This is a good idea but it won't work today. Also, it's not about testing one thing or the other. We should test both against a real RPC Server and `soroban serve` (which unfortunately doesn't work today AFAICR since it doesn't accept contract creation transactions).

Reply:
It feels like we are conflating integration and unit test concerns in a single repo , ideally should be able to trigger the same coverage over functional code from either test scope, unit tests w/mocking or integration. But integration tests can happen less often(nightly, ad-hoc, PR merge) and from a separate dedicated 'qa' repo, where they spin up all the live architecture rather than duplicating that boilerplate code in each functional component repo like cli, env. 

as part of soroban manual e2e testing, had spun out an issue to think about integration tests from an external repo, initially created [#375](https://github.com/stellar/quickstart/issues/375) in quickstart, but it really could be a separate integration test repo and keep out of quickstart which is meant for other purposes.

Reply:
@sreuland I agree, but I think there's more than one layer of integration testing for us to consider so I think we should do that, plus some other things locally in this repo. For example:

1. Integration testing all the way through to the soroban-rpc server, that includes a stellar-core captive-core, maybe we can call that end-to-end testing:

   @sreuland If you're saying this type of test should happen in a separate repo, I agree. I think it's going to slow down development in this repo for the reasons described in [comment](https://github.com/stellar/soroban-cli/pull/220#discussion_r998562880). Especially as long as stellar-core is a component these tests are dependent on. 

   @tsachiherman @graydon @ire-and-curses See @sreuland's comment above. We should discuss effort to setup a QA style end-to-end test repo that runs all components of the entire Soroban experience with a test of test vectors as a top-down view of compatibility.

2. Integration testing from the sense of outside-in testing the CLI using real wasm files:

   I think we can do that in this repo. The performance cost is much smaller. Incompatibilities are much easier and faster to resolve. It would be faster to iterate on code protected by these types of tests than the tests in (1). We've found this model work well for the env repo, although I acknowledge it is not perfect.

3. Integration testing from the sense of outside-in with mocked inputs:

   I think we can do that in this repo, although I think (2) may be easier for us to implement in the short term because mocking will make it harder to change the overall design, and with #197 there might be a lot of changes.

---

@2opremio I think it's fine if you want to continue with exploring an integration test in this repo that uses quickstart in the interim until the above idea is worked out, but I think it is important that:
- Any test dependent on a running rpc server should be skipped by default. Typically skipping tests are done via a feature as I describe here https://github.com/stellar/soroban-cli/pull/220#discussion_r998560044.
- That the existing integration tests in the tests/ repo aren't skipped as part of ☝🏻.

Reply:
Regarding @sreuland's idea for a test repo, that's captured in https://github.com/stellar/jump-cannon-deliverables/issues/25 that he just opened recently.